### DISCUSSION TITLE
Count of messages?

### DISCUSSION THREAD

**Discussion by burkgar** [O365/python-o365#639] [Q&A]
Is there a way to get a count of messages within a folder / count of messages returned by a query? I can't seem to find any info on this.

Note: I realize you can convert the generator object to a list and get a count of the list - just wanted to see if there is a better way to get the count.

**Comment by alejcas** [comment]
Folder.total_items_count

Just look a bit into de code.

**Reply by burkgar** [reply]
Thank you for the response. Funny enough I figured it out the day before your post! I also added a quick and dirty method to my mailbox.py file that allows for a count of items returned by a query but it probably needs to be cleaned up.

`def get_message_counts(self, limit=25, *, query=None, order_by=None, batch=None,
                     download_attachments=False):
        """
        Counts messages from this folder

        :param int limit: limits the result set. Over 999 uses batch.
        :param query: applies a filter to the request such as
         "displayName eq 'HelloFolder'"
        :type query: Query or str
        :param order_by: orders the result set based on this condition
        :type order_by: Query or str
        :param int batch: batch size, retrieves items in
         batches allowing to retrieve more items than the limit.
        :param bool download_attachments: whether or not to download attachments
        :return: list of messages
        :rtype: list[Message] or Pagination
        """

        if self.root:
            url = self.build_url(self._endpoints.get('root_messages'))
        else:
            url = self.build_url(self._endpoints.get('folder_messages').format(
                id=self.folder_id))

        if not batch and (limit is None or limit > self.protocol.max_top_value):
            batch = self.protocol.max_top_value

        params = {'$top': batch if batch else limit}

        if query:
            if isinstance(query, str):
                params['$filter'] = query
            else:
                params.update(query.as_params())

        params['$count'] = str('true')
        response = self.con.get(url, params=params)
        if not response:
            return str('No response or bad response.')

        counts = response.json()
        
        return counts['@odata.count']`

**Reply by alejcas** [reply]
If you just want to count better use a query.select to reduce the size of the query

**Reply by burkgar** [reply]
Thank you for the tip - anywhere I can find more info on query.select?

**Reply by alejcas** [reply]
Look at the code. Utils.py i think
